clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -108,10 +108,11 @@ public static ChannelBuilder createChannelBuilder(AbstractConfig config, Time ti
 
     static List<InetAddress> resolve(String host, ClientDnsLookup clientDnsLookup) throws UnknownHostException {
         InetAddress[] addresses = InetAddress.getAllByName(host);
-        if (ClientDnsLookup.USE_ALL_DNS_IPS == clientDnsLookup) {
-            return filterPreferredAddresses(addresses);
-        } else {
+        if (ClientDnsLookup.USE_FIRST_DNS_IP == clientDnsLookup) {
             return Collections.singletonList(addresses[0]);
+        } else {
+            // ClientDnsLookup.USE_ALL_DNS_IPS == clientDnsLookup || ClientDnsLookup.DEFAULT == clientDnsLookup
+            return filterPreferredAddresses(addresses);
         }
```

Comment:
@badaiaqrandista We should replace this if/else with a switch statement so that we are forced to handle every case. That is error prone if we add new elements to the enum. Also, can we please add tests?

Reply:
Btw, it seems that we are changing the behavior of `RESOLVE_CANONICAL_BOOTSTRAP_SERVERS_ONLY` as well, right? Before, we would only use the first DNS IP for that option in the non bootstrap path. It would be good to make that clear in the KIP.

Reply:
Yes, you are correct. If `client.dns.lookup=RESOLVE_CANONICAL_BOOTSTRAP_SERVERS_ONLY` , then it will resolves the hostname to IP address using the default behaviour.

Reply:
I've changed this into switch/case and updated the KIP.

Reply:
Can we remove the default since we handled all the cases?

Reply:
ok. will remove the default.